.travis.yml

```diff
@@ -0,0 +1,43 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+language: cpp
+sudo: required
+
+before_install:
+  - sudo apt-get -qq update
+  - sudo apt-get -y build-essential
+  - sudo apt-get install -y cmake
+  - sudo apt-get install -y doxygen
+  - sudo apt-get install -y git
+  - sudo apt-get install -y openjdk-8-jdk
+  - sudo apt-get install -y wget
+  - sudo apt-get install -y zlib1g-dev
+
+install: mkdir build && cd build && cmake ../src && cmake --build . -- -j 8
```

Comment:
The configuration step will need to be like so: 
`make ../src -DGEODE_ROOT=<pathtogeode> `
optionally one could add:
`-DCMAKE_BUILD_TYPE=Debug` Debug is default unless Release is specified
`-DPRODUCT_VERSION=maj.min.patch-build.N`